### PR TITLE
File Activation changes 

### LINKED ISSUE #705: Ability to open a YAML file directly into the machine configuration yaml flow
### Suggested new feature or improvement

First, need to investigate and confirm usefulness of such a utility. Creating issue as placeholder.

It'd be helpful if a user could quickly either open a YAML file into the machine config flow of Dev Home and go through the configuration steps OR people could link to existing yaml packages that are in Dev Home.

### Scenario

Make it easier to link YAML files to Dev Home machine config flow. Can we make it easier for users who discover configuration files from documentation to quickly get in the Dev Home setup flow?

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by joadoumie**
## Summary of the pull request
Files of type "*.winget" can launch Dev Home into the machine configuration flow.

## Detailed description of the pull request / Additional comments
This PR adds logic into Dev Home to add "*.winget" files as an activation handler so that users can launch directly into the Dev Home machine configuration flow by opening a "*.winget" file with Dev Home.
## Validation steps performed
- Tested launching valid .winget file while Dev Home was closed, and while Dev Home was already open.
- Tested launching invalid .winget file while Dev Home was closed, and while Dev Home was already open.
- Tested launching a file while already in the machine config flow with a different file.
- Tested launching a file while already in the machine config flow with the existing file.

## PR checklist
- [ x ] Closes #705 
- [ x ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

**Comment by joadoumie** [comment]
@guimafelipe - as discussed there is a bug where the flow doesn't work for an invalid .dsc file

**Comment by joadoumie** [comment]
Fixed bug for invalid .dsc file

**Comment by bbonaby** [comment]
@joadoumie  I noticed this has still been in draft for about 2 weeks now. Do you still need a review on this or are you still working on it?

**Comment by joadoumie** [comment]
@bbonaby I was OOF for a while... definitely do want to get this thing reviewed and make the necessary changes to get it merged in.

**Comment by joadoumie** [comment]
@bbonaby I was OOF for a while... definitely do want to get this thing merged in and noticing some conflicts atm.

**Comment by joadoumie** [comment]
Following up on this. 

@denelon for visibility

File: src/ViewModels/ShellViewModel.cs
Comment (EricJohnson327) [inline src/ViewModels/ShellViewModel.cs:60]: This is fine for now, but if we ever handle other file types as well this will need to be updated.

**Review by EricJohnson327 — APPROVED** [review]

File: src/Services/FileActivationHandler.cs
Comment (AmelBawa-msft) [inline src/Services/FileActivationHandler.cs:4]: Some of the `using` can be omitted

File: src/Services/FileActivationHandler.cs
Comment (AmelBawa-msft) [inline src/Services/FileActivationHandler.cs:21]: Nit: Consider renaming this file to be more specific (e.g.: `DSCConfigurationFileActivationHandler`) since the `CanHandleInternal` is domain specific

File: src/Services/FileActivationHandler.cs
Comment (AmelBawa-msft) [inline src/Services/FileActivationHandler.cs:30]: Since FileActivationHandler is a registered type, hence the types explicitly resolved from the container can be injected:
```cs
public FileActivationHandler(SetupFlowViewModel setupFlowViewModel, INavigationService navigationService)
{
    _setupFlowViewModel = setupFlowViewModel;
    _navigationService = navigationService;
}
```

File: src/Services/FileActivationHandler.cs
Comment (AmelBawa-msft) [inline src/Services/FileActivationHandler.cs:35]: nit: last await can be removed

File: src/ViewModels/ShellViewModel.cs
Comment (AmelBawa-msft) [inline src/ViewModels/ShellViewModel.cs:62]: NavigateTo is called in the file handler class, is this line still needed?

Comment (joadoumie) [reply src/ViewModels/ShellViewModel.cs:62]: If you remove that line the behavior I was seeing was that on launch you'd still end up on the dashboard. I was thinking this may be the case if this is called after the activation handling kicks in

**Comment by EricJohnson327** [comment]
Ping @joadoumie Is this still wanted?  Converting to draft until comments are taken care of.

**Comment by joadoumie** [comment]
@EricJohnson327 yes definitely we still want it! 

@AmelBawa-msft was making a bunch of changes though that will impact this flow so I was waiting until those PRs are merged in to work with Amir to add the functionality back in.

**Comment by AmelBawa-msft** [comment]
Closing in favor of #2379

**Pull request closed without merging**
